src/util/string.cpp

```diff
@@ -945,3 +947,22 @@ std::wstring translate_string(const std::wstring &s) {
 	translate_all(s, i, res);
 	return res;
 }
+
+int mt_snprintf(char * const buf, const size_t buf_size, const char *fmt, ...) {
+	// https://msdn.microsoft.com/en-us/library/bt7tawza.aspx
+	//  Many of the MSVC / Windows printf-style functions do not support positional arguments (eg. "%1$s").
+	//  We just forward the call to vsnprintf for sane platforms, but defer to _vsprintf_p on MSVC / Windows.
+	// https://github.com/FFmpeg/FFmpeg/blob/5ae9fa13f5ac640bec113120d540f70971aa635d/compat/msvcrt/snprintf.c#L46
+	//  _vsprintf_p has to be shimmed with _vscprintf_p on -1 (for an example see above FFmpeg link).
+	va_list args;
+	va_start(args, fmt);
+#ifndef _WIN32
```

Comment:
Should be `_MSC_VER`
https://github.com/minetest/minetest/pull/7308#discussion_r186995693